### ISSUE TITLE
rebar size

### ISSUE THREAD

**Issue by VartuloFN** [terminatorul/NvStrapsReBar#24]
Motherboard model: MSI Z390-A PRO (MS-7B98)
    Motherboard chipset: Intel Z390 (Cannon Lake-H)
    CPU Model: Intel Core i7-9700
    Graphic card: PNY GTX 1660 Super
    VRAM size: 6144MB (6GB)
    New BAR size (GPU-Z): 2048MB (2GB)
    New BAR size (nvidia-smi): 2048MB
    NVIDIA driver version: 551.52
![GPU-z](https://github.com/terminatorul/NvStrapsReBar/assets/137735109/35e04118-be6f-4264-9c09-8eb7f1ea5431)
![gpu-z1](https://github.com/terminatorul/NvStrapsReBar/assets/137735109/fc39f990-7892-4890-a76e-f6c7788ea28b)
![image](https://github.com/terminatorul/NvStrapsReBar/assets/137735109/5e45c02c-2932-4b9d-95e9-eb11086c4b36)

_Originally posted by @VartuloFN in https://github.com/terminatorul/NvStrapsReBar/issues/1#issuecomment-1958025149_
            
            Is it fine to increase the rebar size manually to 8GB?

**Comment by pabloalber84** [comment]
In fact the correct thing to do (if im wrong, anyone correct me), is to assign a space in Rebar to the nearest amount of memory.
Memory ranges vary between:
`16MB > 32MB > 64MB > 128MB > 256MB > 512MB > .... > 4GB > 8GB > 16GB > 32GB > ...`
If you look, there is a jump from 4GB to 8GB, so you will have to allocate the closest (upward) which is 8GB. If your graphics card had 12GB of VRAM for example, you would have to allocate 16GB, because from the 8GB it jumps to 16GB the correct ReBar space.
Obviously, they recommend having at least twice as much RAM as you are allocated in ReBar.

**Comment by xCuri0** [comment]
@pabloalber84 the ram thing isn't true


You just need to set it to the size which covers all your GPU VRAM

**Comment by terminatorul** [comment]
@VartuloFN 
>        Is it fine to increase the rebar size manually to 8GB?

Yes, this is the recommended thing to do.

The NvStrapsReBar DXE driver identifies the card by the PCI ID, so that way it can tell how much memory the GPU should have. But there are many possible ID values (about 130, you can check source file DeviceRegistry.c if you want) and it is possible some of them are missing, which is what happend with your card (PCI IDs 10DE:21C4 196E:1345). 2 GiB size is just the default value if the card is not identified.

So yes, go ahead and manually configure the card to 8 GiB BAR size. Now that I have the PCI ID, a future version of NvStrapsReBar will fix this issue as well.

**Comment by VartuloFN** [comment]
Yeah I already did that 2 days ago i set the size manually to 8GB , yes works fine .
Just i asked because the automatic option have set it to me to 2GB and it’s to low compared to my gpu vram size.

**Comment by VartuloFN** [comment]
And what do you recommend to set rebar size limit In NvidiaProfileInspector to?

**Comment by terminatorul** [comment]
Most game titles reach best performance around 1.5 GiB BAR size, which is what NVIDIA uses for most titles. But not all titles are like this, some games benefit from setting the rBAR size to maximum (8 GiB) in the Profile Inspector. I do not have examples other then 3DMark PCI bandwidth test, so you should try for yourslef

**Comment by VartuloFN** [comment]
Yeah i will do tests of course , the hexaDecimal values is represents the rebar size in bytes?

**Comment by terminatorul** [comment]
Yes
